lib/services/server.go

```diff
@@ -428,3 +433,41 @@ func NodeHasMissedKeepAlives(s types.Server) bool {
 	serverExpiry := s.Expiry()
 	return serverExpiry.Before(time.Now().Add(apidefaults.ServerAnnounceTTL - (apidefaults.ServerKeepAliveTTL() * 2)))
 }
+
+// NewAWSNodeFromEC2Instance creates a Node resource from an EC2 Instance.
+// It has a pre-populated spec which contains info that is not available in the ec2.Instance object.
+func NewAWSNodeFromEC2Instance(instance ec2Types.Instance, awsCloudMetadata *types.AWSInfo) (types.Server, error) {
+	labels := libaws.TagsToLabels(instance.Tags)
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+	libaws.AddMetadataLabels(labels, awsCloudMetadata.AccountID, awsCloudMetadata.Region)
+
+	awsCloudMetadata.InstanceID = aws.ToString(instance.InstanceId)
+	awsCloudMetadata.VPCID = aws.ToString(instance.VpcId)
+
+	if aws.ToString(instance.PrivateIpAddress) == "" {
+		return nil, trace.BadParameter("private ip address is required from ec2 instance")
+	}
+	// Address requires the Port.
+	// We use the default port for the OpenSSH daemon.
+	addr := net.JoinHostPort(aws.ToString(instance.PrivateIpAddress), "22")
```

Comment:
Nit: Do we have a constant for the default ssh port? If not, can you make one?

Reply:
Different question, can this port be chnaged? In almost all cases, AFAIK, this can be changed in `sshd`, and there is always someone who does that.

Reply:
I thought about the same reviewing this, I think we can keep this scenario out of scope for now, we'd probably have to take the port from some tag/label on the instance if someone is using a custom port.

Reply:
Moved to a const.

This can be any value, but we default to 22 because, well, it's the default and we don't know where the OpenSSH server is listening on.
Another thing, is that the EC2 Instance Connect Endpoint only supports connections targeting the port 22 (or 3389/rdp)
So, if the OpenSSH server is running somewhere else, we wouldn't be able to connect to this resource.